### PR TITLE
Fix ID validation failure when browser is refreshed

### PR DESCRIPTION
**Pull request by DinikaSen**
### Proposed changes in this pull request

This PR introduces the following changes.
- Fix ID validation failures that occur when the browser is refreshed after login into the sample app.
- Fix internal server error message shown in Tomcat instead of showing the specific error via the error page of the sample.
- Fix error shown in Tomcat when trying to logout from the SAML sample app from two logged in browser tabs

Should be merged after https://github.com/asgardeo/asgardeo-java-saml-sdk/pull/22 is merged and a new version of the SDK is released.

### CODE REVIEW

File: io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java [hunk 1]
```diff
@@ -86,76 +86,122 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
...
+                    return;
+                }
                 return;
             } else if (resolver.isSAML2SSOResponse()) {
 
-                samlSSOManager = new SAML2SSOManager(ssoAgentConfig);
                 try {
+                    samlSSOManager = new SAML2SSOManager(ssoAgentConfig);
                     samlSSOManager.processResponse(request, response);
+                    if (!"true".equals(request.getAttribute(SSOAgentConstants.SHOULD_GO_TO_WELCOME_PAGE))) {
```
Comment (ChamathNS) [inline io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java:104]: Shall we use Parse boolean here?

File: io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java [hunk 2]
```diff
@@ -86,76 +86,122 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
...
+                    if (resolver.isHttpPostBinding()) {
+
+                        boolean isPassiveAuth = ssoAgentConfig.getSAML2().isPassiveAuthn();
+                        ssoAgentConfig.getSAML2().setPassiveAuthn(false);
+                        String htmlPayload = samlSSOManager.buildPostRequest(request, response, true);
+                        ssoAgentConfig.getSAML2().setPassiveAuthn(isPassiveAuth);
+                        SSOAgentUtils.sendPostResponse(request, response, htmlPayload);
+
+                    } else {
+                        //if "SSOAgentConstants.HTTP_BINDING_PARAM" is not defined, default to redirect
```
Comment (ChamathNS) [inline io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java:140]: let's fix the comment here as well :)

File: io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java [hunk 3]
```diff
@@ -86,76 +86,122 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
...
-                    //if "SSOAgentConstants.HTTP_BINDING_PARAM" is not defined, default to redirect
-                    boolean isPassiveAuth = ssoAgentConfig.getSAML2().isPassiveAuthn();
-                    ssoAgentConfig.getSAML2().setPassiveAuthn(false);
-                    String redirectUrl = samlSSOManager.buildRedirectRequest(request, true);
-                    ssoAgentConfig.getSAML2().setPassiveAuthn(isPassiveAuth);
-                    response.sendRedirect(redirectUrl);
+                try {
+                    samlSSOManager = new SAML2SSOManager(ssoAgentConfig);
+                    if (resolver.isHttpPostBinding()) {
+
```
Comment (ChamathNS) [inline io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java:132]: let's remove the blank line

**Review by ChamathNS — APPROVED** [review]

File: io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java (hunk 1 above)
Comment (DinikaSen) [reply io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java]: Addressed with https://github.com/asgardeo/asgardeo-tomcat-saml-agent/pull/47/commits/dee9d739015733c812fbefca25ffab28bc6b2ffd

File: io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java (hunk 2 above)
Comment (DinikaSen) [reply io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java]: Addressed with https://github.com/asgardeo/asgardeo-tomcat-saml-agent/pull/47/commits/dee9d739015733c812fbefca25ffab28bc6b2ffd

File: io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java (hunk 3 above)
Comment (DinikaSen) [reply io.asgardeo.tomcat.saml.agent/src/main/java/io/asgardeo/tomcat/saml/agent/SAML2SSOAgentFilter.java]: Addressed with https://github.com/asgardeo/asgardeo-tomcat-saml-agent/pull/47/commits/dee9d739015733c812fbefca25ffab28bc6b2ffd

**Pull request merged**